core/src/test/scala/shapeless/annotation.scala

```diff
@@ -175,4 +188,30 @@ class AnnotationTests {
     illTyped(" TypeAnnotations[Second, Dummy] ", "could not find implicit value for parameter annotations: .*")
   }
 
+  @Test
+  def allAnnotations: Unit = {
+    {
+      val all: (First :: HNil) :: shapeless.HNil :: (Second :: Third :: HNil) :: HNil = AllAnnotations[CC3].apply()
```

Comment:
You can use `typed` instead of testing this twice

Reply:
First time using `typed`. Did some changes but not sure if this is what you have in mind.